src/lib/middleware/with-session-auth.ts

```diff
@@ -0,0 +1,120 @@
+import jwt from "jsonwebtoken"
+import {
+  HttpException,
+  type Middleware,
+  NotFoundException,
+  UnauthorizedException,
+} from "nextlove"
+import type { AuthenticatedRequest } from "src/types/authenticated-request.ts"
+
+import type { Database } from "lib/database/index.ts"
+
+import { withSimulatedOutage } from "./with-simulated-outage.ts"
+
+export const withSessionAuth =
+  <RequiresWorkspaceId extends boolean>({
+    require_workspace_id,
+  }: {
+    require_workspace_id: RequiresWorkspaceId
+  }): Middleware<
+    {
+      auth: Extract<
+        AuthenticatedRequest["auth"],
+        {
+          type: RequiresWorkspaceId extends true
+            ? "user_session"
+            : "user_session_without_workspace"
+        }
+      >
+    },
+    {
+      db: Database
+    }
+  > =>
+  (next) =>
+  async (req, res) => {
+    const token = req.headers.authorization?.split("Bearer ")?.[1]
+
+    if (token == null) return res.status(401).end("Unauthorized")
+
+    const workspace_id_from_header = req.headers["seam-workspace"]
+    const workspace_id =
+      workspace_id_from_header !== null &&
+      typeof workspace_id_from_header !== "undefined" &&
```

Comment:
This is actually the simplest way to do safe null checks
```suggestion
      workspace_id_from_header != null
```

(or `workspace_id_from_header == null` for the inverse)